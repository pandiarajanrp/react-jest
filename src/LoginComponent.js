import React, { Component } from 'react';
import { Button, Form, FormGroup, Label, Input, FormText } from 'reactstrap';

class LoginComponent extends Component {
    constructor(props) {
        super(props);
        this.state = {
            username: "",
            password: "",
            usernameError: "",
            passwordError: ""
        }
        this.handleChange = this.handleChange.bind(this);
        this.handleSubmit = this.handleSubmit.bind(this);
    }

    handleChange(e) {
        const {name, value} = e.target;
        const stateCopy = Object.assign({}, this.state);
        if (value) {
            stateCopy[name] = value;
            stateCopy[`${name}Error`] = "";
        } else {
            stateCopy[`${name}Error`] = "";
        }
        
        stateCopy[name] = value;
        this.setState(stateCopy);
    }

    handleSubmit(e) {
        e.preventDefault();
        if (!this.isValid()) {
            return false;
        }
    }

    isValid() {
        if (!this.state.username) {
            this.setState({usernameError: "This field is required"});
            return false;
        } else if(!this.state.password) {
            this.setState({passwordError: "This field is required"});
            return false;
        }
        return true;
    }

    render() {
        const {username, password, usernameError, passwordError} = this.state;
        return(
            <div className="col-md-4 login-container">
                <Form onSubmit={this.handleSubmit}>
                    <FormGroup>
                        <Label for="exampleEmail">Email</Label>
                        <Input type="text" name="username" id="username" placeholder="Username" onChange={this.handleChange} value={username}/>
                        {usernameError ? <span className="error">{usernameError}</span> : null}
                    </FormGroup>
                    <FormGroup>
                        <Label for="examplePassword">Password</Label>
                        <Input type="password" name="password" placeholder="Password" onChange={this.handleChange} value={password}/>
                        {passwordError ? <span className="error">{passwordError}</span> : null}
                    </FormGroup>
                    <Button id="btn-submit">Submit</Button>
                </Form>
            </div>
        )
    }
}

export default LoginComponent;