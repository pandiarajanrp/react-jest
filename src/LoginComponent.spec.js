import React from 'react';
import { shallow } from 'enzyme';

import LoginComponent from './LoginComponent';

const wrapper = shallow(<LoginComponent />);

const initialState = {
    username: "",
    password: "",
    usernameError: "",
    passwordError: ""
};

describe('LoginComponent Render', () => {
    it('should render login component', () => {
        expect(wrapper.find('.login-container').length).toEqual(1);
    });
});

describe('LoginComponent UserInteractions', () => {
    it('should set username error state', () => {
        wrapper.find('#btn-submit').simulate('click', {preventDefault: () => {}});
        expect(wrapper.state().usernameError).toEqual("This field is required");
    });
    it('username change will update data in state', () => {
        wrapper.find('#username').simulate('change', {target: { name: "username", value:"testusername"} });
        expect(wrapper.state().username).toEqual("testusername");
    });
});