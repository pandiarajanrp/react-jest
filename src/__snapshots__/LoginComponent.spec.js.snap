// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginComponent UserInteractions should render correctly in "debug" mode 1`] = `
<div
  className="col-md-4"
>
  <Form
    tag="form"
  >
    <FormGroup
      tag="div"
    >
      <Label
        for="exampleEmail"
        tag="label"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        Email
      </Label>
      <Input
        name="username"
        onChange={[Function]}
        placeholder="Username"
        type="email"
        value=""
      />
    </FormGroup>
    <FormGroup
      tag="div"
    >
      <Label
        for="examplePassword"
        tag="label"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        Password
      </Label>
      <Input
        name="password"
        onChange={[Function]}
        placeholder="Password"
        type="password"
        value=""
      />
    </FormGroup>
    <Button
      color="secondary"
      tag="button"
    >
      Submit
    </Button>
  </Form>
</div>
`;

exports[`LoginComponent should render correctly in "debug" mode 1`] = `
<div
  className="col-md-4"
>
  <Form
    tag="form"
  >
    <FormGroup
      tag="div"
    >
      <Label
        for="exampleEmail"
        tag="label"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        Email
      </Label>
      <Input
        id="exampleEmail"
        name="email"
        onChange={[Function]}
        placeholder="with a placeholder"
        type="email"
      />
    </FormGroup>
    <FormGroup
      tag="div"
    >
      <Label
        for="examplePassword"
        tag="label"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        Password
      </Label>
      <Input
        id="examplePassword"
        name="password"
        onChange={[Function]}
        placeholder="password placeholder"
        type="password"
      />
    </FormGroup>
    <Button
      color="secondary"
      tag="button"
    >
      Submit
    </Button>
  </Form>
</div>
`;
